### ISSUE TITLE
cy.screenshot(). Different behavior in headless mode and without it.

### ISSUE THREAD

**Issue by dvkruchinin** [cypress-io/cypress#15605]
### Current behavior
Screenshots taken in non-headless mode differ from screenshots in headless mode. In headless mode, the element is not fully captured.

Non-headless:
![topview_before_wheel](https://user-images.githubusercontent.com/33020454/111953092-3115ce80-8af7-11eb-8389-27f419a29b6f.png)
Headless:
![topview_before_wheel](https://user-images.githubusercontent.com/33020454/111953121-3d9a2700-8af7-11eb-9f8c-4007ee8dfcb5.png)

### Desired behavior
The screenshot completely captures the element. This looks true in non-headless mode.

### Test code to reproduce
```git clone https://github.com/manasars/cvat.git && cd ./cvat```
```git checkout cvat-3D-M4```
```docker-compose -f docker-compose.yml -f docker-compose.dev.yml build```
```docker-compose up -d```
```cd ./tests && npm ci```
```npx cypress run --env coverage=false --headless --browser chrome --spec="cypress/integration/canvas3d_functionality/case_56_canvas3d_functionality_basic_actions.js"```
``` npx cypress open``` and run test ```canvas3d_functionality/case_56_canvas3d_functionality_basic_actions.js```

### Versions
Cypress 6.4.0, 6.8.0
Chrome 89

**Comment by dvkruchinin** [comment]
The display is also different if open the app through cypress and without it on Linux.
Via Cypress:
![topview_before_wheel](https://user-images.githubusercontent.com/33020454/111959534-a5547000-8aff-11eb-8071-8cf073616656.png)
The point cloud is very hard to see.
Just through the browser in the same way as in the screenshot above without headless mode.

**Comment by jennifer-shehane** [comment]
Following the instructions given gives a 401 error when logging in with the tests. Please provide all steps necessary to reproduce.

![Canvas 3D functionality  Basic actions -- Check existing of elements -- after all hook (failed)](https://user-images.githubusercontent.com/1271364/112026710-dd15e480-8b0c-11eb-9570-2f043198c770.png)

**Comment by dvkruchinin** [comment]
Apologize. Also need to add a user:
``` docker exec -it cvat bash -ic 'python3 ~/manage.py createsuperuser' ``` for Linux
```
or
```winpty docker exec -it cvat bash -ic 'python3 ~/manage.py createsuperuser'``` for Windows
name: admin
e-mail: <empty>
password: 12qwaszx
```

**Comment by jennifer-shehane** [comment]
It looks like more of a difference between `cypress run` and `cypress open` mode than the actual browser, whether it is headless of headed. I can recreate this with the repo provided.

**Comment by viktorgogulenko** [comment]
Any updates here? We are experiencing the same issue in Cypress version 10.10.0

**Comment by steveosoule** [comment]
Not sure if I was experiencing the same exact issue as the original post, but I was also having differences between `npx cypress run` and `npx cypress open`.

My problem was that `npx cypress run` was using the Electron browser and I was using `npx cypress open` with Chrome. When I changed to using `npx cypress run --browser chrome`, I began to experience consistent results between `run` and `open`.

Additionally, it also helped me to get more consistent results by using this config adjustment to created consistent viewports between `run` and `open`: https://docs.cypress.io/api/plugins/browser-launch-api#Set-screen-size-when-running-headless
